cmd/ocm/create/cluster/cmd.go

```diff
@@ -912,6 +918,22 @@ func promptName(argv []string) error {
 	return fmt.Errorf("A cluster name must be specified")
 }
 
+// promptProvider reads or prompts for the provider
+func promptProvider(fs *pflag.FlagSet, connection *sdk.Connection) error {
+	// Only offer the 2 providers known to support OSD now;
+	// but don't validate if set, to not block `ocm` CLI from creating clusters on future providers.
+	providers, _ := osdProviderOptions(connection)
+
+	err := arguments.PromptOneOf(fs, "provider", providers)
+	if err != nil {
+		return err
+	}
+	if args.provider == "" {
```

Comment:
Given that provider is inferred for the case where subscription is gcpMarketPlace,
 i.e provider  is only required if subscription is not gcpMarketPlace,  
We cannot mark the provider flag as required (the Cmd.MarkFlagRequired)  and instead we are doing a  manual check  if !gcpMarketPlace